src/middlewared/middlewared/alembic/versions/21.MM/2021-05-25_22-00_afp_to_smb.py

```diff
@@ -0,0 +1,162 @@
+"""Convert AFP shares to SMB shares
+
+Revision ID: 5f4cab067712
+Revises: bd637e18fb0b
+Create Date: 2021-05-25 19:00:34.776266+00:00
+
+"""
+import json
+import textwrap
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '5f4cab067712'
+down_revision = 'bd637e18fb0b'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table('sharing_cifs_share', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('cifs_timemachine_quota', sa.Integer(), nullable=True))
+
+    op.execute("UPDATE sharing_cifs_share SET cifs_timemachine_quota = 0")
+
+    with op.batch_alter_table('sharing_cifs_share', schema=None) as batch_op:
+        batch_op.alter_column('cifs_timemachine_quota', existing_type=sa.INTEGER(), nullable=False)
+
+    conn = op.get_bind()
+    has_cifs_home = bool(conn.execute("SELECT * FROM sharing_cifs_share WHERE cifs_home = 1"))
+    disable_acl_if_trivial = []
+    for share in conn.execute("SELECT * FROM sharing_afp_share").fetchall():
+        cifs_auxsmbconf = [
+            "fruit:locking = netatalk",
+            "fruit:metadata = netatalk",
+            "fruit:resource = file",
+            "streams_xattr:prefix = user.",
+            "streams_xattr:store_stream_type = no",
+            "oplocks = no",
+            "level2 oplocks = no",
+            "strict locking = auto",
+        ]
+        share_disable_acl_if_trivial = False
+        if share["afp_allow"].strip():
+            cifs_auxsmbconf.append(f"valid users = {share['afp_allow'].strip()}")
+            share_disable_acl_if_trivial = True
+        if share["afp_deny"].strip():
+            cifs_auxsmbconf.append(f"invalid users = {share['afp_deny'].strip()}")
+            share_disable_acl_if_trivial = True
+        if share["afp_ro"].strip():
+            cifs_auxsmbconf.append(f"rolist = {share['afp_ro'].strip()}")
+            share_disable_acl_if_trivial = True
+        if share["afp_rw"].strip():
+            cifs_auxsmbconf.append(f"rwlist = {share['afp_rw'].strip()}")
+            share_disable_acl_if_trivial = True
+        umask_inv = "%o" % (~int(share["afp_umask"], 8) & 0o777)
+        cifs_auxsmbconf.append(f"create mask = 0{umask_inv}")
+        cifs_auxsmbconf.append(f"directory mode = 0{umask_inv}")
+        cifs_auxsmbconf.append(f"force create mask = 0{share['afp_fperm']}")
+        cifs_auxsmbconf.append(f"force directory mode = 0{share['afp_dperm']}")
+        if share["afp_auxparams"].strip():
+            cifs_auxsmbconf.append(textwrap.indent(f"\nNetatalk Auxiliary Parameters:\n\n{share['afp_auxparams']}", "; "))
+
+        cifs_share = {
+            "cifs_purpose": "MULTI_PROTOCOL_AFP",
```

Comment:
Let's do NO_PRESET here and use the legacy_afp_compatibility checkbox instead.

Reply:
@themylogin are we addressing this review too?

Reply:
@william-gr sorry, discussion still pending in Slack

Reply:
My recommendation was to remove the MULTI_PROTOCOL_AFP preset and migrate shares with this as well to the new column for cifs_legacy_afp_compat. Then apply the relevant parameters to registry based on this config.

Reply:
Keeping / using a purpose that's explicitly related to AFP support seems to be odd.